app/services/survey_response_processor.rb

```diff
@@ -40,9 +59,44 @@ def ak_action
     @ak_action ||= ActionRepository.get(@params[:meta][:action_id])
   end
 
+  # AK's API has an inconsistent format when it comes to creating vs updating actions with custom fields.
+  # * On creation: To pass a custom field it must be passed as a regular field prepending the `action_*` prefix.
+  # example: { action_my_field: 'hello world' }
+  # * On update: To pass a custom field it must be passed as a json object on the `fields` key.
+  # example: { fields: { my_field: 'hello world' }
+  #
+  # We're also prepending the `survey_` prefix to all custom fields.
+
   def update_params
-    @params[:params].clone.tap do |p|
-      p[:page] = @ak_action[:page_ak_id]
+    params = @params[:params].clone
+    params[:page] = ak_action[:page_ak_id]
+
+    # Move all params prefixed with `action_` to
+    # the fields key and remove the prefix
```

Comment:
How come you need to both remove the `action_` prefix, and below, replace the `action_` prefix with `action_survey_`?